geopandas/io/tests/test_geoarrow.py

```diff
@@ -0,0 +1,95 @@
+import io
+import urllib
+
+import pytest
+
+import pyarrow as pa
+from pyarrow import feather
+import geopandas
+
+
+def assert_table_equal(left, right, check_metadata=False):
+    if left.equals(right, check_metadata=check_metadata):
+        return
+
+    if not left.schema.equals(right.schema):
+        raise AssertionError(
+            "Schema not equal\nLeft:\n{0}\nRight:\n{1}".format(
+                left.schema, right.schema
+            )
+        )
+
+    if check_metadata:
+        if not left.schema.equals(right.schema, check_metadata=True):
+            if not left.schema.metadata == right.schema.metadata:
+                raise AssertionError(
+                    "Metadata not equal\nLeft:\n{0}\nRight:\n{1}".format(
+                        left.schema.metadata, right.schema.metadata
+                    )
+                )
+        # TODO also check field metadata
+        for col in left.schema.names:
+            assert left.schema.field(col).equals(
+                right.schema.field(col), check_metadata=True
+            )
+
+    for col in left.column_names:
+        a_left = pa.concat_arrays(left.column(col).chunks)
+        a_right = pa.concat_arrays(right.column(col).chunks)
+        if not a_left.equals(a_right):
+            raise AssertionError(
+                "Column '{0}' not equal:\n{1}".format(col, a_left.diff(a_right))
+            )
+
+    raise AssertionError("Tables not equal for unknown reason")
+
+
+@pytest.mark.parametrize(
+    "geometry_type",
+    [
+        pytest.param("point", marks=pytest.mark.xfail),
+        "linestring",
+        "polygon",
+        "multipoint",
+        "multilinestring",
+        "multipolygon",
+    ],
+)
+def test_geoarrow_export(geometry_type):
+    base_url = (
+        "https://raw.githubusercontent.com/geoarrow/geoarrow-data/v0.1.0/example/"
+    )
```

Comment:
Right now, this test is using the online data, while I think _ideally_ for testing we would have this data locally (to avoid requiring network access for too many tests that are not specific about network access).

The problem is that all those small files together still become a 2.5MB, which is probably too much to add for our binaries. But so adding them in the repo then first needs some setup to ensure we exclude the test data from the wheel, and still gracefully skip those tests if the data is not available (or use network access as backup).

Reply:
Is a git submodule out of the question?

Reply:
Well, I could also use the WKB arrow version instead of GPKG. That should already read fine with our current `read_feather`. 
I initially thought to use something more independent as the base source, but in the end when it goes through a GeoDataFrame, it looses all arrow metadata anyway, so that should still be a fine base data to test `to_arrow()`.

Reply:
When not including the gpkg files, it's only 56kb, so I just included those files in the repo.